### ISSUE TITLE
@RequestParam not working in 1.5.6, was working in 1.5.4

### ISSUE THREAD

**Issue by fawzyj** [spring-projects/spring-boot#9954]
@RequestParam MVC binding is not working in 1.5.6, was working in 1.5.4

This code is working ok in 1.5.4, 

`@RequestMapping(method = PUT, path = "/{collectionId}/products")
    public String addProducts(@ModelAttribute("collection") Collection collection, @RequestParam(value = "product_ids", required = false) List<String> products, Model model, RedirectAttributes ra) {
........`

In 1.5.6 `products` is null, while in 1.5.4 products is a List instance containing valid product IDs

Not sure if this is Spring boot related or Spring Framework, so I posted here

Thanks

**Comment by wilkinsona** [comment]
Can you please provide a small sample that reproduces the problem?

**Comment by jhoeller** [comment]
Could this be related to https://jira.spring.io/browse/SPR-15828?

**Comment by wilkinsona** [comment]
Thank you, @jhoeller. This does indeed appear to be related to SPR-15828.

@fawzyj Thanks for the sample. It works as it did with 1.5.4 with Spring Boot 1.5.6 if I override `spring.version` to use 4.3.11 snapshots:

```groovy
buildscript {
    ext {
        springBootVersion = '1.5.6.RELEASE'
    }
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
    }
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'

version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
    mavenCentral()
    maven { url 'https://repo.spring.io/libs-snapshot' }
}

ext['spring.version'] = '4.3.11.BUILD-SNAPSHOT'

dependencies {
    compile('org.springframework.boot:spring-boot-starter-thymeleaf')
    compile('org.springframework.boot:spring-boot-starter-web')
    testCompile('org.springframework.boot:spring-boot-starter-test')
}
```

I'll close this as it's a duplicate of an issue that's already been fixed in Spring Framework 4.3.11. We'll pick up 4.3.11 for our 1.5.7 release.

**Comment by fawzyj** [comment]
OK, Thanks
